owncloudApp/src/test/java/com/owncloud/android/presentation/viewmodels/settings/SettingsVideoUploadsViewModelTest.kt

```diff
@@ -0,0 +1,286 @@
+    fun `get video uploads path - ok - load before`() {
+        every { preferencesProvider.getString(any(), any()) } returns examplePath
+
+        videosViewModel.loadVideoUploadsPath()
+        val uploadPath = videosViewModel.getVideoUploadsPath()
+
+        assertEquals(examplePath, uploadPath)
+    }
+
+    @Test
+    fun `load video uploads source path - ok`() {
+        mockkStatic(PreferenceManager.CameraUploadsConfiguration::class)
+
+        every { preferencesProvider.getString(any(), any()) } returns exampleSourcePath
+        // It has to be "" for the test to pass
+        every { PreferenceManager.CameraUploadsConfiguration.getDefaultSourcePath() } returns ""
+
+        videosViewModel.loadVideoUploadsSourcePath()
+
+        verify(exactly = 1) {
+            preferencesProvider.getString(
+                PreferenceManager.PREF__CAMERA_VIDEO_UPLOADS_SOURCE,
+                PreferenceManager.CameraUploadsConfiguration.getDefaultSourcePath()
+            )
+        }
+    }
+
+    @Test
+    fun `get video uploads source path - ko - no load before`() {
+        val uploadSourcePath = videosViewModel.getVideoUploadsSourcePath()
+
+        assertNull(uploadSourcePath)
+    }
+
+    @Test
+    fun `get video uploads source path - ok - load before`() {
+        mockkStatic(PreferenceManager.CameraUploadsConfiguration::class)
+
+        every { preferencesProvider.getString(any(), any()) } returns exampleSourcePath
+        every { PreferenceManager.CameraUploadsConfiguration.getDefaultSourcePath() } returns ""
+
+        videosViewModel.loadVideoUploadsSourcePath()
+        val uploadSourcePath = videosViewModel.getVideoUploadsSourcePath()
+
+        assertEquals(exampleSourcePath, uploadSourcePath)
+    }
+
+    @Test
+    fun `handle select video uploads path - ok`() {
+        val data: Intent = mockk()
+        val ocFile: OCFile = mockk()
+
+        every { ocFile.remotePath } returns exampleRemotePath
+        every { data.getParcelableExtra<OCFile>(any()) } returns ocFile
+
+        videosViewModel.handleSelectVideoUploadsPath(data)
+
+        verify(exactly = 1) {
+            data.getParcelableExtra<OCFile>(UploadPathActivity.EXTRA_FOLDER)
+            ocFile.remotePath
+            preferencesProvider.putString(PreferenceManager.PREF__CAMERA_VIDEO_UPLOADS_PATH, exampleRemotePath)
+        }
+    }
+
+    @Test(expected = NullPointerException::class)
+    fun `handle select video uploads path - ko - data intent is null`() {
+        videosViewModel.handleSelectVideoUploadsPath(null)
+    }
+
+    @Test(expected = NullPointerException::class)
+    fun `handle select video uploads path - ko - folder to upload is null`() {
+        val data: Intent = mockk()
+
+        every { data.getParcelableExtra<OCFile>(any()) } returns null
+
+        videosViewModel.handleSelectVideoUploadsPath(data)
+
+        verify(exactly = 1) {
+            data.getParcelableExtra<OCFile>(UploadPathActivity.EXTRA_FOLDER)
+        }
+    }
```

Comment:
We should not expect a `NullPointerException`.

We are not handling them. It will crash the app 🤯